src/games/strategy/engine/framework/ui/NewGameChooser.java

```diff
@@ -51,20 +54,29 @@ private NewGameChooser(final Frame owner) {
     setupListeners();
     setWidgetActivation();
     updateInfoPanel();
+    Timer timer = new Timer();
+    timer.schedule(new TimerTask(){
+      @Override
+      public void run() {
+        if(!finishedRefreshing){
+          JOptionPane.showMessageDialog(owner, "Loading the Maps took more than 5 seconds... \n Thanks for your patience.");
```

Comment:
Not quite on target. This could show while maps are still loading, and secondly, we should be telling our users what they can do about the situation. Always important to identify what has happened, and what the user can or should do about it.